### DISCUSSION TITLE
Building go binary without glibc dependency

### DISCUSSION THREAD

**Discussion by anandprabhala** [tommie/v8go#72] [Q&A]
Currently our go binary is being built on Ubuntu 24.04 and is getting linked to GLIBC_2.38 and GLIBCXX_3.4.32. This works fine in the ubuntu 24.04 machines and containers but fails on 22.04 and older machines.

Is there a way to build with static linking to glibc?

**Comment by tommie** [comment]
The Linux libraries stored in the repo are built on 22.04: https://github.com/tommie/v8go/actions/runs/11484341410/job/31961886386

(GitHub tried upgrading their runners' default to 24.04, but had to downgrade again.)

These are static libraries, so the choice of dynamic linking happens at binary link time.

Perhaps https://mt165.co.uk/blog/static-link-go/ is helpful?
